src/actions/addPackageJsonFunding.ts

```diff
@@ -0,0 +1,125 @@
+import mls from 'multilines'
+import Octokit, { ReposGetResponse } from '@octokit/rest'
+
+import { base64 } from '../utils'
+import { getCollectiveWithGithubHandle } from '../collective'
+import { resetBranch } from '../github'
+
+import defaultFundingPackageAsString from '../assets/default-funding-package'
+
+const BRANCH_NAME = `opencollective-bot/funding_package`
+
+const FILE_PATH = 'package.json'
+
+const PR_TITLE = 'Add FUNDING to package.json'
+
+const PR_BODY = mls`
+  | Add FUNDING to package.json
+  |
+  | Add FUNDING to package.json
+  |
+  | Add FUNDING to package.json
+  `
+
+const FUNDING_DEFAULT_FILE_CONTENT = defaultFundingPackageAsString
+
+/**
+ * Finds last line before closing curly bracket of package.json
+ * Adds comma
+ * Adds fundingUrl (new properties to package.json)
+ * Add closing curly bracket
+ * @param packageJSON
+ * @param fundingUrl
+ */
+function updatePackageJson(packageJSON: string, fundingUrl: string) {
+  const obj = JSON.parse(packageJSON)
+  obj['funding'] = {
+    type: 'opencollective',
+    url: fundingUrl,
+  }
+  return JSON.stringify(obj, null, 2)
```

Comment:
As part of the issue, it was mentioned to pay extra attention to formatting.

In this case, we will force 2 spaces indentation. If the `package.json` is indented with tabs or 4 spaces this can be a problem. 

We need to detect indentation and respect it, similar to npm: https://github.com/npm/npm/issues/5694 https://github.com/npm/npm/commit/64b67f00